From 3331916a07d137604aa1652d2cd9af5c37a22e22 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 29 Jun 2016 14:29:31 +0300
Subject: [PATCH 0310/1345] net: mvpp2x: fix: disable autoneg for 2.5 Gbits
 port

commit  e103add866e5c9105965fd0cfed5df8cf02b8afd from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- disable PCS and bypass in GMAC for 2.5 Gbits port

Change-Id: I4066cc5f022d0ffaddd87808b80d7f5fb1371b60
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30844
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c |    8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
index 88a636a..aef6281 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
@@ -309,11 +309,9 @@ static void mv_gop110_gmac_sgmii2_5_cfg(struct gop_hw *gop, int mac_num)
 	val |= MV_GMAC_PORT_CTRL0_PORTTYPE_MASK;
 	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL0_REG, val);
 
-	/* configure AN 0x926C */
-	an = MV_GMAC_PORT_AUTO_NEG_CFG_EN_PCS_AN_MASK |
-		MV_GMAC_PORT_AUTO_NEG_CFG_AN_BYPASS_EN_MASK |
-		MV_GMAC_PORT_AUTO_NEG_CFG_SET_MII_SPEED_MASK  |
-		MV_GMAC_PORT_AUTO_NEG_CFG_SET_GMII_SPEED_MASK     |
+	/* configure AN 0x9260 */
+	an = MV_GMAC_PORT_AUTO_NEG_CFG_SET_MII_SPEED_MASK  |
+		MV_GMAC_PORT_AUTO_NEG_CFG_SET_GMII_SPEED_MASK |
 		MV_GMAC_PORT_AUTO_NEG_CFG_ADV_PAUSE_MASK    |
 		MV_GMAC_PORT_AUTO_NEG_CFG_SET_FULL_DX_MASK  |
 		MV_GMAC_PORT_AUTO_NEG_CFG_CHOOSE_SAMPLE_TX_CONFIG_MASK;
-- 
1.7.9.5

